app/models/collection.rb

```diff
@@ -131,11 +131,13 @@ def to_open_search_collections(document, hits, params, url, for_atom = true, cwi
           if create_cwic_osdd_link || has_granules
             link_title = "Granule OpenSearch Descriptor Document"
             if Flipper.enabled?(:use_cwic_server)
-              #Rails.logger.info "Flipper says use cwic server!"
-              add_link_as_child(doc, node, "#{Rails.configuration.cwic_granules_osdd_endpoint}opensearch/datasets/#{id}/osdd.xml?clientId=#{params[:clientId]}", 'application/opensearchdescription+xml', NEW_REL_MAPPING[:search], link_title)
+              href_link = "#{Rails.configuration.cwic_granules_osdd_endpoint}opensearch/datasets/#{id}/osdd.xml"
+              !params[:clientId].blank? ? href_link = href_link + "?clientId=#{params[:clientId]}" : nil
```

Comment:
change this to something like: `href_link = href_link + "?clientId=#{params[:clientId]}" unless params[:clientId].blank?`